packages/components/src/components/post-button/post-button.tsx

```diff
@@ -0,0 +1,243 @@
+   */
+  @Prop() readonly formenctype: null | string = null;
+
+  /**
+   * Defines the HTTP method for the form.
+   */
+  @Prop() readonly formmethod: null | string = null;
+
+  /**
+   * Defines if the form should be validated when submitted.
+   */
+  @Prop() readonly formnovalidate: boolean;
+
+  /**
+   * Defines where to display the response after submitting the form.
+   */
+  @Prop() readonly formtarget: null | string = null;
+
+  /**
+   * Defines the name of the button.
+   */
+  @Prop() readonly name: null | string = null;
+
+  /**
+   * Defines the target of the popover.
+   */
+  @Prop() readonly popovertarget: null | string = null;
+
+  /**
+   * Defines the action of the popover target.
+   */
+  @Prop() readonly popovertargetaction: null | string = null;
+
+  /**
+   * Defines the type of the button.
+   */
+  @Prop() readonly type: null | 'button' | 'reset' | 'submit' = null;
+
+  /**
+   * Defines the value of the button.
+   */
+  @Prop() readonly value: null | string = null;
+
+  constructor() {
+    this.setClasses = this.setClasses.bind(this);
+  }
+
+  @Watch('variant')
+  variantChanged() {
+    this.setClasses();
+  }
+
+  @Watch('size')
+  sizeChanged() {
+    this.setClasses();
+  }
+
+  @Watch('iconOnly')
+  iconChanged() {
+    this.setClasses();
+  }
+
+  @Watch('animated')
+  animatedChanged() {
+    this.setClasses();
+  }
+
+  @Watch('class')
+  classChanged() {
+    this.setClasses();
+  }
+
+  private animationClass() {
+    if (this.animated === 'start') return 'btn-animated-start';
+    if (this.animated === 'end') return 'btn-animated';
+    return null;
+  }
+
+  private setClasses() {
+    this.classes = [
+      'btn',
+      this.animationClass(),
+      this.size ? `btn-${this.size}` : null,
+      this.variant ? `btn-${this.variant}` : null,
+      this.iconOnly ? 'btn-icon' : null,
+      ...this.class?.split(' ').filter(c => c.startsWith('w-')),
+    ]
+      .filter(c => c !== null)
+      .join(' ');
+  }
+
+  connectedCallback() {
+    this.setClasses();
+  }
+
+  render() {
+    const startSection = this.getStartSection();
+    const endSection = this.getEndSection();
+
+    return (
+      <Host data-version={version}>
+        <button
+          class={this.classes}
+          disabled={this.disabled || this.loading}
+          autofocus={this.autofocus}
+          form={this.form}
+          formaction={this.formaction}
+          formenctype={this.formenctype}
+          formmethod={this.formmethod}
+          formnovalidate={this.formnovalidate}
+          formtarget={this.formtarget}
+          name={this.name}
+          popoverTarget={this.popovertarget}
+          popoverTargetAction={this.popovertargetaction}
+          type={this.type}
+          value={this.value}
+        >
+          {startSection}
+          <span class={this.iconOnly ? 'visually-hidden' : undefined}>
+            <slot />
+          </span>
+          {endSection}
+        </button>
```

Comment:
Generally, it would be nice to treat the `post-button` as the button instead of nesting a button inside. I expect this to simplify things because attributes added to the custom element don't need to be passed down to the button. What we need to make sure then is, that the custom button plays nice with all the attributes that can be added to real buttons.

For forms, for example, this means adding form association (https://www.raymondcamden.com/2023/05/24/adding-form-participation-support-to-web-components) which gives access to the form the button is part of and enables us to react to `type="reset"` to clear the form. This would not work in the current implementation, because of the shadow boundary (https://codepen.io/tuelsch/pen/NWZvEQw, only the reset light deletes the text in the input field). The reset shadow button has no form association, but is inside the shadow dom.